src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/InheritDocTaglet.java

```diff
@@ -36,70 +39,80 @@
+ * declarations and from overridden and implemented methods.
  */
 public class InheritDocTaglet extends BaseTaglet {
 
     /**
      * Construct a new InheritDocTaglet.
      */
-    public InheritDocTaglet () {
+    public InheritDocTaglet() {
         super(DocTree.Kind.INHERIT_DOC, true, EnumSet.of(Location.TYPE, Location.METHOD));
     }
 
     /**
      * Given an element, a {@code DocTree} in the element's doc comment
-     * replace all occurrences of @inheritDoc with documentation from its
+     * replace all occurrences of {@code {@inheritDoc}} with documentation from its
      * superclass or superinterface.
      *
      * @param writer the writer that is writing the output.
      * @param e the {@link Element} that we are documenting.
-     * @param holderTag the tag that holds the inheritDoc tag or null for type
-     * (class) docs.
-     * @param isFirstSentence true if we only want to inherit the first sentence.
+     *
+     * @param holderTag
+     *
+     * either the tag that holds the {@code {@inheritDoc}} tag or {@code null},
+     * which can mean either of:
+     * <ul>
+     *     <li>the tag is used on a class {@link jdk.javadoc.doclet.Taglet.Location#TYPE} declaration, or
+     *     <li>the tag is used to copy the overall doc comment
+     * </ul>
+     *
+     * @param isFirstSentence true if we only want to inherit the first sentence
      */
     private Content retrieveInheritedDocumentation(TagletWriter writer,
-            Element e, DocTree holderTag, boolean isFirstSentence) {
+                                                   Element e,
+                                                   DocTree holderTag,
+                                                   boolean isFirstSentence) {
         Content replacement = writer.getOutputInstance();
         BaseConfiguration configuration = writer.configuration();
         Messages messages = configuration.getMessages();
         Utils utils = configuration.utils;
         CommentHelper ch = utils.getCommentHelper(e);
-        Taglet inheritableTaglet = holderTag == null
+        Taglet taglet = holderTag == null
                 ? null
                 : configuration.tagletManager.getTaglet(ch.getTagName(holderTag));
-        if (inheritableTaglet != null &&
-            !(inheritableTaglet instanceof InheritableTaglet)) {
-                String message = utils.getSimpleName(e) +
-                    ((utils.isExecutableElement(e))
-                        ? utils.flatSignature((ExecutableElement)e, writer.getCurrentPageElement())
-                        : "");
-                //This tag does not support inheritance.
-                var path = writer.configuration().utils.getCommentHelper(e).getDocTreePath(holderTag);
-                messages.warning(path, "doclet.inheritDocWithinInappropriateTag", message);
-                return replacement;
+        if (taglet != null && !(taglet instanceof InheritableTaglet)) {
```

Comment:
Optional, this is a possible use case for patterns in `instanceof` ... you could declare a name here and save the cast on line 95

Reply:
I think `taglet != null` gets in the way. I tried to change the order of conjuncts, but to no avail: flow scoping analysis seems to give up. Might be something that the compiler folks can comment on.

Reply:
I think `taglet != null` gets in the way. I tried to change the order of conjuncts, but to no avail: flow scoping analysis seems to give up. Might be something that the compiler folks can comment on.

Reply:
Noted